src/callbacks_step/analysis.jl

```diff
@@ -401,6 +406,34 @@ function (analysis_callback::AnalysisCallback)(io, du, u, u_ode, t, semi)
     end
   end
 
+  # Total Variation
+  if :total_variation in analysis_errors
+    if mpi_isroot()
+      print(" TV(U):  ")
+      @assert ndims(equations) == 1 "Total Variation is only applicable for 1D"
+      @assert semi.boundary_conditions == 
+              boundary_condition_periodic "Total Variation is currently implemented for periodic B.C. only"
+
+      @assert nvariables(equations) == 1 "Total Variation is currently implemented for scalars only"
+      # TODO: Non-scalar implementation!
+      for v in eachvariable(equations)
+        TV = 0
+        for element_index in 1:nelements(semi.cache.elements) - 1
+          TV += abs(integrate_element(cons2cons, u_ode, element_index + 1, semi)[v] - 
+                    integrate_element(cons2cons, u_ode, element_index,     semi)[v])
```

Comment:
That would be quite confusing for me.

- On a uniform mesh, this is essentially "take the TV seminorm of the underlying FV method". 
- However, our numerical solution itself is also a vector, so you could take `sum(abs, diff(u_ode))` as TV seminorm.
- Even more confusing, our numerical solution should be a polynomial on each element, so you could (in principle) compute its [total variation](https://en.wikipedia.org/wiki/Bounded_variation).

We should hear what @trixi-framework/principal-developers think about this.

Reply:
>     On a uniform mesh, this is essentially "take the TV seminorm of the underlying FV method".

Yes, this is what I had in mind when constructing this. In fact, I am mostly interested in the special case of the TVD constant polynomial, first order Finite Volume case discussed in [another issue](https://github.com/trixi-framework/Trixi.jl/issues/1318).
In that case computing the integrals is gross overkill - something like `sum(abs, diff(u_ode))` should do the job, yes.

In any case, I am very interested in your opinions on this and what notion you consider generally meaningful.

Reply:
Well, I am honestly not sure since there are different conflicting notions of possibly reasonable definitions. That's why I asked the others :sweat_smile: